crates/rattler_lock/src/conda.rs

```diff
@@ -1,145 +1,116 @@
+    pub(crate) channel: Option<Url>,
+}
 
-    /// Experimental: see: [Track features](rattler_conda_types::PackageRecord::track_features)
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    #[serde_as(as = "OneOrMany<_>")]
-    pub track_features: Vec<String>,
+impl AsRef<PackageRecord> for CondaPackageData {
+    fn as_ref(&self) -> &PackageRecord {
+        &self.package_record
+    }
+}
 
-    /// Experimental: the specific license of the package
-    pub license: Option<String>,
+impl PartialOrd<Self> for CondaPackageData {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
 
-    /// Experimental: the license family of the package
-    pub license_family: Option<String>,
+impl Ord for CondaPackageData {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.package_record
+            .name
+            .cmp(&other.package_record.name)
+            .then_with(|| {
+                self.package_record
+                    .version
+                    .cmp(&other.package_record.version)
+            })
+            .then_with(|| self.package_record.build.cmp(&other.package_record.build))
+            .then_with(|| self.package_record.subdir.cmp(&other.package_record.subdir))
+    }
+}
 
-    /// Experimental: If this package is independent of architecture this field specifies in what way. See
-    /// [`NoArchType`] for more information.
-    #[serde(skip_serializing_if = "NoArchType::is_none")]
-    pub noarch: NoArchType,
+impl CondaPackageData {
+    /// Returns the filename of the package.
+    pub fn file_name(&self) -> Option<&str> {
+        self.file_name
+            .as_deref()
+            .or_else(|| file_name_from_url(&self.url))
+    }
 
-    /// Experimental: The size of the package archive in bytes
-    pub size: Option<u64>,
+    /// Returns the channel of the package.
+    pub fn channel(&self) -> Option<Url> {
+        self.channel.clone().or_else(|| channel_from_url(&self.url))
+    }
+}
 
-    /// Experimental: The date this entry was created.
-    #[serde_as(as = "Option<crate::utils::serde::Timestamp>")]
-    pub timestamp: Option<chrono::DateTime<chrono::Utc>>,
+impl From<RepoDataRecord> for CondaPackageData {
+    fn from(value: RepoDataRecord) -> Self {
+        let derived_file_name = file_name_from_url(&value.url);
+        let file_name = if derived_file_name == Some(value.file_name.as_str()) {
+            None
+        } else {
+            Some(value.file_name)
+        };
 
-    /// Experimental: Defines that the package is an alias for a package from another ecosystem.
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    pub purls: Vec<PackageUrl>,
+        Self {
+            package_record: value.package_record,
+            url: value.url,
+            file_name,
+            // TODO: This is not entirely correct. It should be derived from the `channel` field in
```

Comment:
Yes, so the `channel` field should be `Some` if it can't be properly derived from the URL of the package. But the channel in `RepoDataRecord` at this point is some undefined String. I left it like this until we get around to fix the `channel` field in `RepoDataRecord`.